_posts/blog/2018-01-24-modernizing-mlab.md

```diff
@@ -0,0 +1,83 @@
+
+Upgrades to nearly all locations in the United States were completed by mid-2017. Additionally, over 30 new sites in Europe have been added in 2017, and M-Lab operations is now beginning hardware upgrades for our sites elsewhere in the world. Where possible, our new sites in Europe will have 10 Gb connectivity. We will seek to upgrade our existing site hardware in the United States and Europe beginning in late fall 2017.
+
+### Open Source ETL Pipeline
+
+In late summer 2017, we moved to a new open source data acquisition and ETL pipeline for ingesting data into Google Cloud Storage and BigQuery. The pipeline code for [acquisition](https://github.com/m-lab/scraper) and [ETL](https://github.com/m-lab/etl) is available in Github and is now in production for NDT. It currently supports geographic annotations and will support ASN annotation in a few months. Our development team is currently experimenting with new efficient schemas for NDT and Paris Traceroute. Look for more on the open source ETL pipeline transition in an upcoming blog post.
+
+### Monitoring and Alerting
+
+M-Lab operations currently uses several tools to monitor our infrastructure, and to alert us when something needs to be addressed. Nagios and the other open source tools we use for monitoring have been our mainstay for several years, but many have inadequate visualization and alerting capabilities. We’ve begun transitioning our monitoring and alerting platform to [Prometheus](http://prometheus.io/), which will provide monitoring for all of our sites, switches, machines, containers, and servers. As of this writing, we are running Prometheus in parallel with Nagios, and development is proceeding to transition from Nagios entirely in early 2018.
+
+### Remote Boot Management and Server Provisioning
+
+The design process and initial development of [ePoxy](https://github.com/m-lab/epoxy), a safe network bootloader and API, began in 2015. ePoxy is a replacement for the Planet Lab network bootloader used by M-Lab to load authenticated operating systems and virtualization containers securely across the Internet. Our development team has confirmed that ePoxy can be used to deploy an M-Lab server’s current operating system (Centos) and vserver virtualization containers. We also tested deployment of a container running [CoreOS](https://coreos.com/), which M-Lab intends to use as the base operating system on our servers after this platform upgrade is completed. The move to ePoxy will remove M-Lab’ dependency on Planet Lab and therefore addresses one reason that M-Lab remains on an older Linux kernel.
```

Comment:
Please change to `The move to ePoxy will remove M-Lab's boot-time dependency on Planet Lab and therefore addresses one reason that M-Lab remains on an older Linux kernel. We anticipate continuing to use PlanetLab's identity-management systems on our updated platform.`